src/main/java/uk/gov/pay/adminusers/service/InviteService.java

```diff
@@ -141,16 +151,32 @@ public void generateOtp(InviteOtpRequest inviteOtpRequest) {
     }
 
     @Transactional
-    public User createInvitedUser(InviteValidateOtpRequest inviteValidateOtpRequest) {
+    public User createInvitedUserAndInvalidateInvite(InviteValidateOtpRequest inviteValidateOtpRequest) {
         Optional<InviteEntity> inviteOptional = inviteDao.findByCode(inviteValidateOtpRequest.getCode());
         if (inviteOptional.isPresent()) {
-            if (!secondFactorAuthenticator.authorize(inviteOptional.get().getOtpKey(), inviteValidateOtpRequest.getOtpCode())) {
-                throw invalidOtpAuthCodeInviteException(inviteValidateOtpRequest.getCode());
-            }
             InviteEntity inviteEntity = inviteOptional.get();
+            if (!secondFactorAuthenticator.authorize(inviteEntity.getOtpKey(), inviteValidateOtpRequest.getOtpCode())) {
+                // "failed login attempt" logic
+                inviteEntity.setLoginCounter(inviteEntity.getLoginCounter() + 1);
+                inviteEntity.setDisabled(inviteEntity.getLoginCounter() >= loginAttemptCap);
+                inviteDao.merge(inviteEntity);
+                // throw UNAUTHORIZED / 401 exception if the otp validation failed
+                throw invalidOtpAuthCodeInviteException(inviteEntity.getCode());
```

Comment:
As discussed with @kakumara and @rauligs we could use http://www.vavr.io/ library and http://www.vavr.io/vavr-docs/#_either functionality to fix that.
Later, we decided to create temporary class which will be removed after refactoring and the need to use "Either" behaviour.